pci/src/vfio.rs

```diff
@@ -716,6 +721,61 @@ impl VfioCommon {
         }
     }
 
+    // PCI Express Spec 4.0r1.0 section 7.6.1
+    // Parse the PCI extended capabilities present in PCI Express
+    // and explicity allow capabilities supported by cloud-hypervisor,
+    // This function builds a map of the dword headers by index, such that
+    // further reads on these index values in read_shadow_dword() return a PCI
+    // capabilities header potentially with offset modified.
+    //
+    // PCI Express Capability headers are read-only as per specification.
+    fn parse_extended_capabilities(&mut self) {
+        let mut header_offset = PCI_CONFIG_EXTENDED_CAPABILITY_OFFSET;
+        while {
+            let header = self.vfio_wrapper.read_config_dword(header_offset);
+            let header_cap: u16 = (header & 0x0000_ffff).try_into().unwrap();
+            let cap: PciExpressCapabilityId = header_cap.into();
+            info!(
+                "Capability {:#06x}@{:#06x} = {:?}",
+                header_cap, header_offset, cap
+            );
+            header_offset /= 4;
+
+            //
+            // Only match PCIE capabilities explicitly tested with real hardware
+            // If your VFIO device has an unlisted capability that you
+            // would like to enable, please submit a PR. Many capabilities
+            // could be added to this allowlist without direct cloud-hypervisor
+            // support, although what these capabilites are, are unknown.
+            match cap {
+                PciExpressCapabilityId::VirtualChannelMultiFunctionVirtualChannelNotPresent => {
+                    self.write_shadow_dword(header_offset.try_into().unwrap(), header);
```

Comment:
read if there is a way to just list these matches as a group, so there is no need to cargo-cult-copy the write_shadow_dword() call.